### ISSUE TITLE
Vignettes are not included in package and source links lead to 404s

### ISSUE THREAD

**Issue by JedGrabman** [cmu-delphi/covidcast#102]
The [Get started with covidcast](https://cmu-delphi.github.io/covidcast/covidcastR/articles/covidcast.html) guide includes a link to the source at vignettes/covidcast.Rmd. This is likely intended to lead to:
https://github.com/cmu-delphi/covidcast/blob/main/R-packages/covidcast/vignettes/covidcast.Rmd

but actually leads to the non-existent: 
https://github.com/cmu-delphi/covidcast/blob/main/vignettes/covidcast.Rmd
and results in a 404.

While the vignette is viewable online, I followed the instructions at the [covidcast R package documentation](https://cmu-delphi.github.io/covidcast/covidcastR/) and found that I could not reference the vignettes from RStudio i.e. vignette("covidcast") does not work.

This appears to affect other vignette links as well.

Based on the format of other packages, it looks like a covidcast/Meta/vignette.rds file is missing, as well as covidcast/doc, so there may be an issue with the package configuration.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1198968030437421/1198970311860844) by [Unito](https://www.unito.io/learn-more)

**Comment by capnrefsmmat** [comment]
It looks like `devtools::install_github` might not build and install the vignettes by default. They do take a long time to build. I wonder if we should find another way to distribute the package (or work towards getting on CRAN properly).

As for the link not working, I'd appreciate help figuring that out. The source links are broken on all the documentation pages, because pkgdown seems to assume the package always lives in the root of the GitHub repository. I didn't figure out how to change that, but if you discover a setting I've overlooked, we should definitely fix it.

**Comment by JedGrabman** [comment]
I was able to build the vignettes by running the following

```
devtools::install_github("cmu-delphi/covidcast", ref = "main",
                         subdir = "R-packages/covidcast", 
                         build_vignettes = TRUE,
                         dependencies = TRUE)
```
It's probably worth mentioning this as an option in the documentation, although it is quite slow.

I'm not very familiar with pkgdown. I did see that [the documentation](https://pkgdown.r-lib.org/reference/build_site.html#yaml-config) differentiates between a destination (where the site is generated) and the url (where site is hosted). We [set the destination](https://github.com/cmu-delphi/covidcast/blob/main/R-packages/covidcast/_pkgdown.yml):

```
# Deploy to central docs directory that's on GitHub Pages
destination: ../../docs/covidcastR/
```
It may be that because the destination is under cmu-delphi/covidcast, it believes that that is the root of the package and that setting the URL instead would fix the issue, but I haven't tested it.

**Comment by capnrefsmmat** [comment]
The destination ensures that when I'm in `R-packages/covidcast/` and run `pkgdown::build_site()`, it places the rendered documentation in `docs/covidcastR/` instead of in `R-packages/covidcast/docs/`. Placing them under `docs/` is necessary for our GitHub Pages setup.

It seems that interacts badly with its package auto-linking.

**Comment by JedGrabman** [comment]
I spent a little bit trying to find a way to fix the 404's but didn't come up with anything simple. While we potentially could generate files then move them, I think going down that route would be fragile and wouldn't recommend it.

Probably best to file a bug report with pkgdown since it seems to generate the links incorrectly when using the destination flag.
I can do that and add to our installation documentation to explain how they can build vignettes if they want to spend extra time.

**Comment by capnrefsmmat** [comment]
That would be great, thanks. pkgdown probably needs an option to specify where the package is relative to the GitHub repo root, so it can generate the links accordingly.

**Comment by JedGrabman** [comment]
I've created a pkgdown bug [here](https://github.com/r-lib/pkgdown/issues/1420).

**Comment by JedGrabman** [comment]
pkgdown bug was closed and [pointed us back to using repo](https://github.com/r-lib/pkgdown/issues/1420#issuecomment-721381153).

**Comment by JedGrabman** [comment]
No longer getting 404s, this looks fixed by #281.
